crates/router/src/core/webhooks.rs

```diff
@@ -201,6 +208,130 @@ pub async fn payments_incoming_webhook_flow(
     }
 }
 
+#[cfg(feature = "payouts")]
+pub async fn payouts_incoming_webhook_flow(
+    state: SessionState,
+    merchant_account: domain::MerchantAccount,
+    business_profile: diesel_models::business_profile::BusinessProfile,
+    key_store: domain::MerchantKeyStore,
+    webhook_details: api::IncomingWebhookDetails,
+    event_type: webhooks::IncomingWebhookEvent,
+    source_verified: bool,
+) -> CustomResult<WebhookResponseTracker, errors::ApiErrorResponse> {
+    metrics::INCOMING_PAYOUT_WEBHOOK_METRIC.add(&metrics::CONTEXT, 1, &[]);
+    if source_verified {
+        let db = &*state.store;
+        //find refund by connector refund id
+        let payout_attempt = match webhook_details.object_reference_id {
+            webhooks::ObjectReferenceId::PayoutId(payout_id_type) => match payout_id_type {
+                webhooks::PayoutIdType::PayoutAttemptId(id) => db
+                    .find_payout_attempt_by_merchant_id_payout_attempt_id(
+                        &merchant_account.merchant_id,
+                        &id,
+                        merchant_account.storage_scheme,
+                    )
+                    .await
+                    .change_context(errors::ApiErrorResponse::WebhookResourceNotFound)
+                    .attach_printable("Failed to fetch the payout attempt")?,
+                webhooks::PayoutIdType::ConnectorPayoutId(id) => db
+                    .find_payout_attempt_by_merchant_id_connector_payout_id(
+                        &merchant_account.merchant_id,
+                        &id,
+                        merchant_account.storage_scheme,
+                    )
+                    .await
+                    .change_context(errors::ApiErrorResponse::WebhookResourceNotFound)
+                    .attach_printable("Failed to fetch the payout attempt")?,
+            },
+            _ => Err(errors::ApiErrorResponse::WebhookProcessingFailure)
+                .attach_printable("received a non-payout id when processing payout webhooks")?,
+        };
+
+        let payouts = db
+            .find_payout_by_merchant_id_payout_id(
+                &merchant_account.merchant_id,
+                &payout_attempt.payout_id,
+                merchant_account.storage_scheme,
+            )
+            .await
+            .change_context(errors::ApiErrorResponse::WebhookResourceNotFound)
+            .attach_printable("Failed to fetch the payout")?;
+
+        let payout_attempt_update = PayoutAttemptUpdate::StatusUpdate {
+            connector_payout_id: payout_attempt.connector_payout_id.clone(),
+            status: common_enums::PayoutStatus::foreign_try_from(event_type)
+                .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
+                .attach_printable("failed payout status mapping from event type")?,
+            error_message: None,
+            error_code: None,
```

Comment:
We need to capture error_code, error_message at connector level and update them here